src/daft-parquet/src/file.rs

```diff
@@ -95,6 +102,214 @@ where
     }
 }
 
+fn resolve_dtype_recursively(
```

Comment:
The `resolve_*_recursively` code is pretty messy. Happy to take on work to convert it into a Visitor pattern if reviewers think its necessary.